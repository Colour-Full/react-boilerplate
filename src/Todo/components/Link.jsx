// @flow

import React, {type Element, type ChildrenArray} from 'react'
import Button from '@atlaskit/button'

export type LinkProps = {
  active: boolean,
  children: ChildrenArray<string>,
  onClick: (any) => void
}

const Link = ({ active, children, onClick }: LinkProps): Element<any> => (
  <Button
    onClick={onClick}
    isSelected={active}
    style={{
      marginLeft: '4px'
    }}
  >
    {children}
  </Button>
)

export default Link
