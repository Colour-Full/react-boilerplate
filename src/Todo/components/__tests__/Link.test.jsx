// @flow

import React from 'react'
import Link from '../Link'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Button from '@material-ui/core/Button'

Enzyme.configure({ adapter: new Adapter() })

describe('A Link Component ', () => {
  it('It should render without throwing an error ', () => {
    const props = {
      active: true,
      children: [],
      onClick: jest.fn()
    }
    const wrapper = shallow(<Link {...props} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should call onClick handler when button is clicked', () => {
    const props = {
      active: true,
      children: [],
      onClick: jest.fn()
    }
    const wrapper = shallow(<Link {...props} />)
    wrapper.find(Button).simulate('click')
    expect(props.onClick).toHaveBeenCalledTimes(1)
  })
})
