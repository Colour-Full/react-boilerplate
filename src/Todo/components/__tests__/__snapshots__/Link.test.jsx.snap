// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Link Component  It should render without throwing an error  1`] = `
<ForwardRef(WithAnalyticsContext(WithAnalyticsEvents(WithDeprecationWarnings(Button))))
  isSelected={true}
  onClick={[MockFunction]}
  style={
    Object {
      "marginLeft": "4px",
    }
  }
/>
`;
